slosh_test_lib/src/docs/legacy.rs

```diff
@@ -0,0 +1,697 @@
+    let mut status_entries = Vec::new();
+    for (old_sym_name, will_implement, notes) in metadata {
+        let (impl_status, form_name, notes) = match (will_implement, notes.is_empty()) {
+            (true, true) => { // will implement and name is/has not yet been changed.
+                if slosh_forms.contains(*old_sym_name) {
+                    (ImplStatus::Implemented, old_sym_name.to_string(), "".to_string())
+                } else {
+                    (ImplStatus::NotYetImplemented, old_sym_name.to_string(), "".to_string())
+                }
+            }
+            (true, false) => { // will implement but with alternate name
+                if slosh_forms.contains(*notes) {
+                    (ImplStatus::Implemented, old_sym_name.to_string(), format!("Renamed to: `{}`", notes) )
+                } else {
+                    (ImplStatus::NotYetImplemented, old_sym_name.to_string(), "".to_string())
+                }
+            }
+            (false, true) => { // will not implement and explanation is NOT provided
+                (ImplStatus::WillNotImplement, old_sym_name.to_string(), "".to_string())
+            }
+            (false, false) => { // will not implement and explanation is provided
+                (ImplStatus::WillNotImplement, old_sym_name.to_string(), notes.to_string())
+            }
+        };
+        status_entries.push(StatusEntry::new(impl_status, form_name, notes));
+    }
+    for s in status_entries.iter() {
+        match s.resolved {
+            ImplStatus::NotYetImplemented => {
+                yet_to_be_implemented += 1;
+            }
+            ImplStatus::Implemented |
+            ImplStatus::WillNotImplement => {
+                completed += 1;
+            }
+        }
+    }
+    Ok(StatusReport {
+        completed,
+        yet_to_be_implemented,
+        status_entries,
+    })
+}
+
+pub fn get_legacy_sl_sh_form_syms(vm: &mut SloshVm, _registers: &[Value]) -> VMResult<Value> {
+    let v = full_legacy_sl_sh_forms_metadata();
+    let v = v
+        .iter()
+        .map(|x| {
+            let name = x.0;
+            vm.alloc_string_ro(name.to_string())
+        })
+        .collect();
+    Ok(vm.alloc_vector(v))
+}
+
+/// Array of triples:
+///     1. String name of the sl-sh form
+///     2. Whether or not it will be implemented in slosh
+///     3.  If 2 is true:
+///             an optional non-null value listing the name the function is mapped to in slosh if
+///             the name has changed.
+///         If 2 is false:
+///             a note a helpful explanation as to why the function is no longer needed or will not be implemented.
+/// ** all values in .0 (or .3 if .2 is true) are assumed to match to some function in (get-globals-sorted) any missing
+///     values for elements where .2 is true are assumed to be unimplemented.
+pub(crate) fn full_legacy_sl_sh_forms_metadata() -> &'static [(&'static str, bool, &'static str)] {
```

Comment:
can start to build up a list, I imagine a lot of these we do not need to implement. but we should explain why or show which new function it's mapped to so we can check it off the list.

Reply:
this list is def incomplete